From 60d454062110fada86db08620ffdb4370595456b Mon Sep 17 00:00:00 2001
From: Octavian Purdila <octavian.purdila@nxp.com>
Date: Wed, 22 Feb 2017 17:36:53 +0200
Subject: [PATCH 1434/5242] MLK-13869 ARM: imx7d: run pm init during
 intialization

commit  39475f0311114d7ba303404ccca785b0ef3d32da from
https://source.codeaurora.org/external/imx/linux-imx.git

This patch adds code that was missed during the rebase of the imx7 PM
patches.

Signed-off-by: Octavian Purdila <octavian.purdila@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/mach-imx/common.h     |    1 +
 arch/arm/mach-imx/mach-imx7d.c |    4 ++++
 2 files changed, 5 insertions(+)

diff --git a/arch/arm/mach-imx/common.h b/arch/arm/mach-imx/common.h
index 9ddc782..697cc9c8ad 100644
--- a/arch/arm/mach-imx/common.h
+++ b/arch/arm/mach-imx/common.h
@@ -207,6 +207,7 @@ static inline void imx7ulp_suspend(void __iomem *ocram_vbase) {}
 void imx6sx_pm_init(void);
 void imx6ul_pm_init(void);
 void imx6ull_pm_init(void);
+void imx7d_pm_init(void);
 void imx7ulp_pm_init(void);
 void imx7ulp_enable_nmi(void);
 void imx6q_pm_set_ccm_base(void __iomem *base);
diff --git a/arch/arm/mach-imx/mach-imx7d.c b/arch/arm/mach-imx/mach-imx7d.c
index 681ea66..2e639e6 100644
--- a/arch/arm/mach-imx/mach-imx7d.c
+++ b/arch/arm/mach-imx/mach-imx7d.c
@@ -130,12 +130,15 @@ static void __init imx7d_init_machine(void)
 	if (parent == NULL)
 		pr_warn("failed to initialize soc device\n");
 
+	of_platform_default_populate(NULL, NULL, parent);
+	imx7d_pm_init();
 	imx_anatop_init();
 	imx7d_enet_init();
 }
 
 static void __init imx7d_init_irq(void)
 {
+	imx_gpcv2_check_dt();
 	imx_init_revision_from_anatop();
 	imx_src_init();
 	irqchip_init();
@@ -169,4 +172,5 @@ static void __init imx7d_map_io(void)
 	.init_machine	= imx7d_init_machine,
 	.init_late	= imx7d_init_late,
 	.dt_compat	= imx7d_dt_compat,
+	.restart        = mxc_restart,
 MACHINE_END
-- 
1.7.9.5

